doc/style_guide.md

```diff
@@ -303,7 +343,19 @@ Godot usage
   `QueueFreeChildren` extension methods.
 
 - DO NOT DISPOSE Godot Node derived objects, call QueueFree or Free
-  instead.
+  instead. Also don't override Dispose in Node derived types, instead
+  use the tree enter and exit callbacks to handle resources that need
+  releasing when removed (unless it is a game entity for which there's
+  a special mechanism, `IEntity` destroyed callbacks)
+
+- Don't use a constructor to setup Godot resources, usually Node
+  derived types should mostly do Godot related, constructor-like
+  operations entirely in _Ready. Many resources are not ready yet when
+  a class is constructed or static variables are being initialized. If
+  this is not followed script variables may not show up correctly in
+  the Godot editor as that relies on creating an instance of the
+  class, and that can fail for example because SimulationParameters
+  are not initialized in the Godot editor.
```

Comment:
We already have some few Node derived classes that uses constructors e.g `DataPoint`, `IconLegend` and `DataLine` among others, although they don't suffer any issues should their setup be moved into _Ready?

Reply:
Yes, preferably if they can. There *probably* is no issue regarding them as they are dynamically created, however any scene attached objects can cause problems due to the code in the constructor running immediately and not when attaching the scene.

Reply:
I think it needs to be mentioned that scene attached nodes are especially vulnerable to script variables failing to show up correctly due to instantiation fail.

Reply:
Does it? This is kind of long already and it's mentioned that script variables don't show up. It's kind of implicit that only Godot scene file contained classes can have script variables showing in the first place. How would you nicely add that bit of info to this?

Reply:
I tried to improve it.